### PR TITLE
Enable prescribed ice for UFS

### PR DESCRIPTION
**Pull request by NickSzapiro-NOAA**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [x] Short (1 sentence) summary of your PR: 
    Enable prescribed ice CICE for UFS via CMEPS driver
- [x] Developer(s): 
    Nicholas Szapiro & Denise Worthen
- [ ] Suggest PR reviewers from list in the column to the right.
- [x] Please copy the PR test results link or provide a summary of testing completed below.
    New ufs-weather-model regression test atm_ds2s_docn_pcice (https://github.com/ufs-community/ufs-weather-model/pull/2186) 
- How much do the PR code changes differ from the unmodified code? 
    - [x] bit for bit (when not in prescribed ice)
    - [ ] different at roundoff level
    - [x] more substantial (when in prescribed ice) 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [x] No
- Does this PR update the Icepack submodule?  If so, the Icepack submodule must point to a hash on Icepack's main branch.
    - [ ] Yes
    - [x] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [x] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [x] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [x] No 
- [x] Please document the changes in detail, including _why_ the changes are made.  This will become part of the PR commit log.
Prescribed ice CICE was enclosed by #ifdef CESMCOUPLED in CMEPS driver. This (with UWM PR) enables the capability to use prescribed ice CICE in ufs-weather-model with CDEPS calls to input from file.

### CODE REVIEW

**Comment by zach1221** [comment]
@BrianCurtis-NOAA @DeniseWorthen could you please review this CICE PR?

File: cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90 [hunk 1]
```diff
@@ -74,13 +56,17 @@ end subroutine ice_prescribed_init
 contains
 !===============================================================================
 
-  subroutine ice_prescribed_init(clock, mesh, rc)
+  subroutine ice_prescribed_init(gcomp, clock, mesh, rc)
 
     ! Prescribed ice initialization
 
-    include 'mpif.h'
+#ifndef SERIAL_REMOVE_MPI
```
Comment (DeniseWorthen) [inline cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90:63]: What is this here for?

File: cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90 [hunk 2]
```diff
@@ -211,6 +207,9 @@ subroutine ice_prescribed_init(clock, mesh, rc)
           hin_max(1) = 999._dbl_kind
        end if
 
+       ! If need initial cice values for coupling
```
Comment (DeniseWorthen) [inline cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90:211]: Do you know this is compatible w/ CESM's use of prescribed ice?

**Review by DeniseWorthen — COMMENTED** [review]
I think the title is very confusing. Can you title it more clearly? You're simply enabling prescribed ice for UFS.

Also, have you reviewed these changes w/ Dave Bailey to ensure that they are compatible w/ CESM's use cases?

File: cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90 (hunk 1 above)
Comment (NickSzapiro-NOAA) [reply cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90:63]: #ifndef SERIAL_REMOVE_MPI follows the usage around use mpi [elsewhere](https://github.com/search?q=repo%3ACICE-Consortium%2FCICE%20SERIAL_REMOVE_MPI&type=code)

The use mpi is for the mpi_bcast below, which I don't think would be needed if the type were implemented in cicecore/cicedyn/infrastructure/comm/mpi/ice_broadcast.F90 too

File: cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90 (hunk 2 above)
Comment (NickSzapiro-NOAA) [reply cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90:211]: I don't. imo, it's a bug not to actually initialize within ice_prescribed_init, especially if coupling sequences can change. So, I decided against bracketing with #ifndef CESMCOUPLED. If needed in CESM, adding into this branch will not change answers

File: cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90 (hunk 1 above)
Comment (DeniseWorthen) [reply cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90:63]: sorry, I'm still not following. You've added a single broadcast_scalar below (for taxmode). What is it about adding that broadcast that requires this ifdef?

File: cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90 (hunk 2 above)
Comment (DeniseWorthen) [reply cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90:211]: W/o knowing this has been tested and confirmed to work for CESM, this should be enclosed w/ ifndef CESMCOUPLED. We should always err on the side of not impacting each other's uses.

File: cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90 (hunk 1 above)
Comment (NickSzapiro-NOAA) [reply cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90:63]: So hopefully better answer...

mpi is needed for an mpi_bcast already there a couple lines below the added broadcast_scalar call

As "use of the mpif.h include file is strongly discouraged and may be deprecated in a future version of MPI" ([docs](https://www.mpi-forum.org/docs/mpi-3.1/mpi31-report/node411.htm)), I updated this section with the same #ifndef SERIAL_REMOVE_MPI "use mpi" as elsewhere. This was the only "include mpif.h" left, afaik

Comment (DeniseWorthen) [reply cicecore/drivers/nuopc/cmeps/ice_prescribed_mod.F90:63]: OK, thanks. I understand.

**Comment by NickSzapiro-NOAA** [comment]
> Also, have you reviewed these changes w/ Dave Bailey to ensure that they are compatible w/ CESM's use cases?

@DeniseWorthen If this is ready for PR to CICE-Consortium, do you want me to do that (now)?

**Comment by jkbk2004** [comment]
@DeniseWorthen @NickSzapiro-NOAA https://github.com/ufs-community/ufs-weather-model/pull/2186 is ready to test. Please, go ahead for final review and approval.

**Review by DeniseWorthen — APPROVED** [review]

**Review by junwang-noaa — APPROVED** [review]

**Comment by FernandoAndrade-NOAA** [comment]
Testing on [#2186](https://github.com/ufs-community/ufs-weather-model/pull/2186) has completed successfully, please continue with merging this PR.

**Pull request merged**